File: PYTHON/PigLatin.py
# challenge link : https://www.codewars.com/kata/520b9d2ad5c005041100000f/train/python

"""
Move the first letter of each word to the end of it, 
then add "ay" to the end of the word. Leave punctuation marks untouched.
"""

import string

def pig_it(text):
    words = text.split(" ")
    word_to_add = "ay"
    return " ".join([word[1:] + word[0] + word_to_add if word not in string.punctuation else word for word in words])



print(pig_it("Hello world !"))
print(string.punctuation)

File: JAVASCRIPT/PigLatin.js
// challenge link : https://www.codewars.com/kata/520b9d2ad5c005041100000f/train/javascript

/*
Move the first letter of each word to the end of it, 
then add "ay" to the end of the word. Leave punctuation marks untouched.
*/

function pigIt(str){
    //Code here
    let tabStr = str.split(" ");
    let punctuation = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~';
    let textToAdd = "ay";
    let result = [];

    for(item of tabStr){
        if(punctuation.includes(item)) result.push(item);
        else result.push(item.slice(1)+item.charAt(0)+textToAdd);
    }
    return result.join(" ");
  }


  console.log(pigIt('Pig latin is cool !'))

  